app/models/user.rb

```diff
@@ -56,6 +59,13 @@ def self.import(file)
         agent.save
       else
         agent.save
+        if agent.is_deleted then 
+          agent.pto_requests.reverse_each do |request|
```

Comment:
possibly query only future requests to cut down on unnecessary looping

Reply:
improved query with agent.pto_requests.where('request_date > ?', Date.today).destroy_all